_rules/form-field-label-descriptive-cc0f0a.md

```diff
@@ -47,16 +47,17 @@ This rule applies to any [programmatic label][] of an element which has one of t
 - `switch`
 - `textbox`
 
-and where both the element and the [programmatic label][] are [visible][].
+## Expectation 1
 
-## Expectation
+Each test target has a [programmatic label][].
```

Comment:
**Issue:** I don't think that works as an expectation, or at least that means the rule doesn't map to 2.4.6 anymore.
Firstly, 2.4.6 does not require label, only that **if the exist** they are descriptive ([Understanding 2.4.6](https://www.w3.org/WAI/WCAG21/Understanding/headings-and-labels.html)):
> This Success Criterion does not require headings or labels. This Success Criterion requires that if headings or labels are provided, they be descriptive.

Secondly, 2.4.6 is about ["WCAG labels"](https://www.w3.org/TR/WCAG21/#dfn-labels) which are not restricted to programmatic ones, here also the Understanding document is explicit:
> This Success Criterion also does not require that content acting as a heading or label be correctly marked up or identified - this aspect is covered separately by [1.3.1: Info and Relationships](https://www.w3.org/WAI/WCAG21/Understanding/info-and-relationships).

---
So, requiring a "WCAG label" is for 3.3.2, requiring that it is correctly marked up is for 1.3.1 (and a bit 4.1.2) and requirinig that it is descriptive is for 2.4.6.

We may be able to have a single rule with secondary requirements, but it seems that several rules for checking these different aspects would be better suited.

Reply:
@Jym77 I am good with removing the expectation if that would fix this. Or another solution that fits better with this is also something I am good with as well.

Reply:
Removing this expectation won't fix it. 2.4.6 is Inapplicable to form fields without label (programmatic or not).

Reply:
I see. As I'm unsure as to my status coming up, I'm good with what the group decides per the conversation.